source/beamline/CLS/CLSJJSlit.h

```diff
@@ -3,166 +3,77 @@
-	void onPVControlConnected();
-	/// to handle the signale of the status PV value changed
-	void onStatusValueChanged();
-
-protected:
-	/// the direction of the blade
-	Direction direction_;
-
-	/// the name of the JJ slit
-	QString name_;
-	/// a brief description of the JJ slit
-	QString description_;
-
-	/// flag of connection
-	bool wasConnected_;
-
-	/// PV Controll for gap PV
-	AMControl *gapPVControl_;
-	/// PV Controll for center PV
-	AMControl *centerPVControl_;
-	/// PV Control for status PV
-	AMControl *statusPVControl_;
-};
-
-/// The class definition of CLSJJSlit, which contains a vertical CLSJJSlitBladesControl and a horizontal CLSJJSlitBladesControl
-class CLSJJSlit : public QObject
-{
-	Q_OBJECT
-
-public:
+	explicit CLSJJSlits(const QString &name, const QString &upperBladePVName, const QString &lowerBladePVName, const QString &inboardBladePVName, const QString &outboardBladePVName, QObject*parent = 0);
 	/// Destructor
-	virtual ~CLSJJSlit();
-	/// Constructor
-	/*!
-	  \param name A unique description of this motor
-	  \param verticalBladesPVBaseName The base of the PV name of the vertical blades (if the motor status was "SMTR16114I1022:status" then the base is "SMTR16114I1022")
-	  \param horizontalBladesPVBaseName The base of the PV name o the horizontal blades (if the motor status was "SMTR16114I1022:status" then the base is "SMTR16114I1022")
-	  \param description A human readable description for this JJ slit
-	  \param parent QObject parent class
-	  */
-	CLSJJSlit(const QString &name, const QString &description, const QString &verticalBladesPVBaseName, const QString &horizontalBladesPVBaseName, double tolerance = AMCONTROL_TOLERANCE_DONT_CARE, double moveStartTimeoutSeconds = 2.0, double limit = 11.0, QObject *parent = 0);
-
-	/// check whether the JJ slit is connected or not
-	bool isConnected() const;
-
-	/// return the limit of the slit
-	double limit() const { return limit_; }
-	/// return the vertical blade control
-	CLSJJSlitBladesControl * verticalBladesControl() const { return verticalBladesControl_; }
-	/// return the horizontal blade control
-	CLSJJSlitBladesControl * horizontalBladesControl() const { return horizontalBladesControl_; }
+	virtual ~CLSJJSlits();
+	/// Returns the name.
+	const QString name() const { return name_; }
+	/// Returns the current connected state.
+	bool isConnected() const { return connected_; }
+
+	/// Returns the upper blade motor control.
+	AMControl* upperBladeControl() const { return upperBlade_; }
+	/// Returns the lower blade motor control.
+	AMControl* lowerBladeControl() const { return lowerBlade_; }
+	/// Returns the inboard blade motor control.
+	AMControl* inboardBladeControl() const { return inboardBlade_; }
+	/// Returns the outboard blade motor control.
+	AMControl* outboardBladeControl() const { return outboardBlade_; }
+
+	/// Returns the vertical blades gap control.
+	CLSJJSlitGapControl* verticalGapControl() const { return verticalGap_; }
```

Comment:
Technically, this can just be an ```AMControl *``` as well.  You can leave it as the more sophisticated subclass if you wish though.  I just wanted to bring it up that it could also be like the above getters.